secretmanager/create_secret.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package secretmanager
+
+// [START secretmanager_create_secret]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1beta1"
+	secretspb "google.golang.org/genproto/googleapis/cloud/secrets/v1beta1"
+)
+
+// createSecret creates a new secret with the given name. A secret is a logical
+// wrapper around a collection of secret versions. Secret versions hold the
+// actual secret material.
+func createSecret(w io.Writer, parent, name string) error {
```

Comment:
IMO - since we deal with both names and ids for these resources, it is clearer to use id in the sample signature, for a value that gets passed to an ID struct member - this is just the final part of the resource name.

Reply:
I don't disagree, but the contributing guide clearly says we should use "name" here: https://github.com/GoogleCloudPlatform/golang-samples/blob/master/CONTRIBUTING.md#common-identifiers

Reply:
That says use `name` for fully qualified names, but this param maps to just the id 'my-secret' - not the name of 'project/my-project/secrets/my-secret', the full resource `name`